loadtest/store/memstore/random.go

```diff
@@ -448,3 +449,70 @@ func (s *MemStore) RandomDraftForTeam(teamId string) (string, error) {
 
 	return draftIDs[rand.Intn(len(draftIDs))], nil
 }
+
+func (s *MemStore) GetRandomScheduledPost() (*model.ScheduledPost, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	// Seed the random generator
+	rand.Seed(time.Now().UnixNano())
+
+	// Check if scheduledPosts is empty
+	if len(s.scheduledPosts) == 0 {
+		return nil, errors.New("no scheduled posts available")
+	}
+
+	// Pick a random index for the outer map
+	randomOuterIndex := rand.Intn(len(s.scheduledPosts))
+	var selectedInnerMap map[string]*model.ScheduledPost
+	outerIndex := 0
+	for _, innerMap := range s.scheduledPosts {
+		if outerIndex == randomOuterIndex {
+			selectedInnerMap = innerMap
+			break
+		}
+		outerIndex++
+	}
+
+	// Check if the selected inner map is empty
+	if len(selectedInnerMap) == 0 {
+		return nil, errors.New("no posts available in selected category")
+	}
+
+	// Pick a random index for the inner map
+	randomInnerIndex := rand.Intn(len(selectedInnerMap))
+	var selectedPost *model.ScheduledPost
+	innerIndex := 0
+	for _, post := range selectedInnerMap {
+		if innerIndex == randomInnerIndex {
+			selectedPost = post
+			break
+		}
+		innerIndex++
+	}
+
+	return selectedPost, nil
+}
+
+func (s *MemStore) DeleteScheduledPost(scheduledPostID string) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	for teamId := range s.scheduledPosts {
+		if _, ok := s.scheduledPosts[teamId][scheduledPostID]; ok {
+			delete(s.scheduledPosts[teamId], scheduledPostID)
+			break
+		}
+	}
+}
```

Comment:
Do we know the team ID when deleting the post? We could skip the outer loop if we do.